x-pack/plugins/maps/public/classes/layers/vector_layer/vector_layer.tsx

```diff
@@ -687,6 +701,29 @@ export class VectorLayer extends AbstractLayer implements IVectorLayer {
     }
   }
 
+  async _syncIsEditable({ syncContext }: { syncContext: DataRequestContext }) {
+    if (syncContext.dataFilters.isReadOnly) {
+      return;
+    }
+    const { startLoading, stopLoading, onLoadError } = syncContext;
+    const dataRequestId = IS_EDITABLE_REQUEST_ID;
+    const requestToken = Symbol(`layer-${this.getId()}-${dataRequestId}`);
+    const prevDataRequest = this.getDataRequest(dataRequestId);
+    if (prevDataRequest) {
+      return;
+    }
+    try {
+      startLoading(dataRequestId, requestToken);
+      const isEditable = await this.getSource().loadIsEditable();
+      stopLoading(dataRequestId, requestToken, { isEditable });
+    } catch (error) {
+      if (!(error instanceof DataRequestAbortError)) {
```

Comment:
You can remove this if check. Since loadIsEditable is not using searchSoruce, you do not need to worry about canceled requests